### PR TITLE
Php Lists added Resolves #289

### LINKED ISSUE #289: Add PHP Lists/Arrays/Hashes
# Description

Lists/Arrays/Hashes for PHP aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/lists.json` to the right PHP directory in `web/thesauruses/`
* Edit the lists.json file to accurately reflect the latest major version of PHP
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by alexindris**
## What GitHub issue does this PR apply to?

Resolves #289 

## What changed and why?

list.json php implementing Arrays l ists and hashes

## Checklist

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

File: web/thesauruses/php/lists.json [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+      "comment": "You can use any of them"
+    },
+    "ordered_mutable_list_start_number": {
+      "code": "0"
+    },
+    "ordered_mutable_list_can_be_appended": {
+      "code": "array_push($array,$value)"
+    },
+    "ordered_mutable_list_can_be_inserted_in_middle": {
+      "code": "",
```
Comment (geekygirlsarah) [inline web/thesauruses/php/lists.json]: For concepts that aren't implemented, can you remove the `code` line? it just helps clean up the file some.

File: web/thesauruses/php/lists.json [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+      "code": "$copy = array_slice($array, $offset, $length); "
+    },
+    "get_list_length": {
+      "code": "count($array)"
+    },
+    "resize_list": {
+      "code": "",
+      "not-implemented": "true"
+    },
+    "do_two_lists_match_exactly": {
```
Comment (geekygirlsarah) [inline web/thesauruses/php/lists.json:337]: doesn't array_diff return an array of different items? I think this (and `do_two_lists_contain_same_items`) are both boolean-based concepts

File: web/thesauruses/php/lists.json [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+      "code": "",
+      "not-implemented": "true"
+    },
+    "does_list_not_satisfy_an_expression": {
+      "code": "",
+      "not-implemented": "true"
+    },
+    "sort_list": {
+      "code": "asort($array, sorttype)",
+      "comment": "Sorttipe is optional\n possible values are\n0 = SORT_REGULAR - Default. Compare items normally (don't change types) \n 1 = SORT_NUMERIC - Compare items numerically\n2 = SORT_STRING - Compare items as strings\n3 = SORT_LOCALE_STRING - Compare items as strings, based on current locale\n4 = SORT_NATURAL - Compare items as strings using natural ordering\n5 = SORT_FLAG_CASE -"
```
Comment (geekygirlsarah) [inline web/thesauruses/php/lists.json]: Did you mean "Sort type is optional"?

File: web/thesauruses/php/lists.json [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+    },
+    "shuffle_list": {
+      "code": "suffle($array)"
```
Comment (geekygirlsarah) [inline web/thesauruses/php/lists.json]: Should this be `shuffle($array)` ?

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
I think with a few changes and fixing some typos, this will be good to merge!

File: web/thesauruses/php/lists.json (hunk 3 above)
Comment (alexindris) [reply web/thesauruses/php/lists.json]: Yes the sort type is optional in order to be able to choose between diferent ways

File: web/thesauruses/php/lists.json (hunk 4 above)
Comment (alexindris) [reply web/thesauruses/php/lists.json]: Yes it is

File: web/thesauruses/php/lists.json (hunk 3 above)
Comment (geekygirlsarah) [reply web/thesauruses/php/lists.json]: I was pointing out the typo, but you got it.

**Comment by geekygirlsarah** [comment]
I think you fixed the issues, so I'll merge this in!

Thank you for adding this! Don't forget to sign up for Hacktoberfest if you haven't already since this counts!

**Pull request merged**